src/lua/spellcheck.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::{anyhow, Result};
+use hunspell_rs::{CheckResult, Hunspell};
+use mlua::prelude::LuaError;
+use mlua::{AnyUserData, Result as LuaResult, String as LuaString, Table, UserData};
+use std::sync::Arc;
+
+pub const LUA_GLOBAL_NAME: &str = "spellcheck";
+
+pub struct Spellchecker {
+    hunspell: Option<HunspellSafe>,
+}
+
+impl Spellchecker {
+    pub fn new() -> Self {
+        Spellchecker { hunspell: None }
+    }
+
+    pub fn init(&mut self, aff_path: &str, dict_path: &str) {
+        self.hunspell
+            .replace(HunspellSafe::from(Hunspell::new(aff_path, dict_path)));
+    }
+
+    fn check_initialized(&self) -> Result<()> {
+        if self.hunspell.is_none() {
+            return Err(anyhow!("spellchecker not initialized"));
```

Comment:
Change to `Err(anyhow!("spellchecker not initialized"))`

You can put the `Ok(())` in an else block or use the `match` pattern.